Slice element pointer replaced by Unmarshal on arm64

**Describe the bug**

When unmarshalling into `[]any` containing a `*[]string` the element pointer (and type) are changed on arm64, but not on amd64, or using golang.org/encoding/json.

**To Reproduce**

```
func TestSlicePointer(t *testing.T) {
	a := []string{}
	b := []any{&a}
	if err := sonic.ConfigStd.Unmarshal([]byte(`[["one","2"]]`), &b); err != nil {
		t.Fatalf("Unmarshal must not error: %s", err)
	}
	if _, ok := b[0].(*[]string); !ok {
		t.Fatalf("b[0] must still a *[]string; Got: %T", b[0])
	}
	if b[0] != &a {
		t.Error("b[0] must still be *a")
	}
}
```
Giving:
```
GOARCH=arm64 go test ./... -run TestSlicePointer
--- FAIL: TestSlicePointer (0.00s)
    sliceptr_test.go:598: b[0] must still a *[]string; Got: []interface {}
```
versus:
```
GOARCH=amd64 go test ./... -run TestSlicePointer
--- PASS: TestSlicePointer (0.01s)
```

**Expected behavior**

The existing slice pointer should be left in place, and the slice updated with the unmarshaled data, as happens on amd64.

**Sonic version:**

v1.12.7

**Environment:**
```
GO111MODULE=''
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/gbjk/Library/Caches/go-build'
GOENV='/Users/gbjk/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/gbjk/go/pkg/mod'
GOOS='darwin'
GOPATH='/Users/gbjk/go'
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.23.2'
GODEBUG=''
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/gbjk/Library/Application Support/go/telemetry'
GCCGO='gccgo'
GOARM64='v8.0'
AR='ar'
CC='clang'
CXX='clang++'
CGO_ENABLED='1'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'

--- a/decode_test.go
+++ b/decode_test.go
@@ -2162,6 +2162,8 @@ func TestSkipArrayObjects(t *testing.T) {
 // slices, and arrays.
 // Issues 4900 and 8837, among others.
 func TestPrefilled(t *testing.T) {
+    var one int = 1
+    var two int = 2
     // Values here change, cannot reuse table across runs.
     var prefillTests = []struct {
         in  string
@@ -2184,6 +2186,11 @@ func TestPrefilled(t *testing.T) {
             out: &[]int{2},
         },
         {
+            in:  `[2]`,
+            ptr: &[]int{1, 3},
+            out: &[]int{2},
+        },
+        {
             in:  `[2, 3]`,
             ptr: &[]int{1},
             out: &[]int{2, 3},
@@ -2198,6 +2205,17 @@ func TestPrefilled(t *testing.T) {
             ptr: &[...]int{1, 2},
             out: &[...]int{3, 0},
         },
+        {
+            in:  `[2]`,
+            ptr: &[]interface{}{&one, 2.0},
+            out: &[]interface{}{&two},
+        },
+        // TODO(rfc): different with encoding/json
+        // {
+        //     in:  `{"a": 2}`,
+        //     ptr: &map[string]interface{}{"a": &one, "b": int(2)},
+        //     out: &map[string]interface{}{"a": 2.0, "b": int(2)},
+        // },
     }
 
     for _, tt := range prefillTests {

--- a/internal/decoder/optdec/slice.go
+++ b/internal/decoder/optdec/slice.go
@@ -95,7 +95,18 @@ func (d *sliceEfaceDecoder) FromDom(vp unsafe.Pointer, node Node, ctx *context) 
 		return nil
 	}
 
-	return node.AsSliceEface(ctx, vp)
+	/* if slice is empty, just call `AsSliceEface` */
+	if ((*rt.GoSlice)(vp)).Len == 0 {
+		return node.AsSliceEface(ctx, vp)
+	}
+	
+	decoder := sliceDecoder{
+		elemType: rt.AnyType,
+		elemDec:  &efaceDecoder{},
+		typ:      rt.SliceEfaceType.Pack(),
+	}
+
+	return decoder.FromDom(vp, node, ctx)
 }
 
 type sliceI32Decoder struct {

--- a/issue_test/common_test.go
+++ b/issue_test/common_test.go
@@ -1,0 +1,31 @@
+// Copyright 2025 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package issue_test
+
+import (
+	"testing"
+
+	"encoding/json"
+	"github.com/bytedance/sonic"
+	"github.com/stretchr/testify/assert"
+)
+
+func assertUnmarshal(t *testing.T, api sonic.API, input []byte, newfn func() interface{}) {
+	sv, jv := newfn(), newfn()
+	serr := api.Unmarshal(input, sv)
+	jerr := json.Unmarshal(input, jv)
+	assert.Equal(t, jv, sv)
+	assert.Equal(t, serr == nil, jerr == nil)
+}

--- a/issue_test/issue750_test.go
+++ b/issue_test/issue750_test.go
@@ -1,0 +1,31 @@
+// Copyright 2025 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package issue_test
+
+import (
+	"testing"
+
+	"github.com/bytedance/sonic"
+)
+
+func genSlice() interface{} {
+	a := []string{}
+	return &[]interface{}{&a}
+}
+
+func TestSlicePointer_Issue750(t *testing.T) {
+	assertUnmarshal(t, sonic.ConfigStd, []byte(`[["one","2"]]`), genSlice)
+
+}

